src/memo/memo.ml

```diff
@@ -146,7 +202,9 @@ module Cached_value = struct
           Fiber.parallel_map acc ~f:Fn.id >>| List.exists ~f:Fn.id
         | Last_dep.T (node, prev_output) :: deps ->
           match node.state with
-          | Running (run, ivar) ->
+          | Running_sync _ ->
+            failwith "Synchronous function called [Cached_value.get_async]"
```

Comment:
Could you use `Exn.code_error`. I don't think we really use `failwith` anywhere in dune.

Reply:
Thanks. Changed.